product_docs/docs/migration_portal/4/03_mp_using_portal/mp_ai_copilot/ai_good_prompts.mdx

```diff
@@ -0,0 +1,108 @@
+    we strongly recommend testing the solutions in a controlled test environment 
+    to ensure the proposed fixes align with your specific migration requirements.
+
+## Asking a general question 
+
+### Neutral prompt
+
+```
+What tools are there to migrate databases? 
+```
+
+Since the AI Copilot is trained on EDB product documentation and knowledge base, this general question will still
+give you an answer in the context of EDB's product offerings. However, the chatbot focusses on the most common use case, 
+migrating from an Oracle database to a Postgres database, which might not be your use case. 
+
+To improve this prompt, be more specific about the source and target databases. 
+
+### Better prompt
+
+```
+What tools does EDB provide to migrate from open-source Postgres to EPAS? 
+```
+
+This prompt will produce a more accurate answer, listing EDB's product offerings that support that specific use case. 
+
+You can then continue the conversation by adding more questions or context, for example: 
+
+```
+What is the difference between the Migration Portal and the Migration Toolkit?
+```
+
+## Finding a solution for a syntax error 
+
+When the Migration Portal finds a syntax issue, it flags a line with the problem.
+
+![Syntax error message](../../images/mp_syntax_error.png)
+
+You can ask a general question to gain more knowledge about the query, for example: 
+
+```
+What are type casts in Postgres? 
+How do I use `SYS.ODCIVARCHAR2LIST`?
+```
+
+You can ask the AI Copilot to correct a query with a syntax error.
+In this case, copy the entire query that contains the line with the issue: 
+
+```
+Can you correct the syntax of this query? `CREATE OR REPLACE VIEW HRPLUS.VW_STRING_LIST (STRING_VAL) AS SELECT COLUMN_VALUE FROM SYS.ODCIVARCHAR2LIST('a','b', 'c');`
+```
+
+Alternatively, you can request an EPAS target equivalent for an Oracle source query:
+
+```
+Can you provide an EPAS-compatible equivalent for `CREATE OR REPLACE FORCE EDITIONABLE VIEW "HRPLUS"."VW_STRING_LIST" ("STRING_VAL") AS SELECT "COLUMN_VALUE" FROM SYS.ODCIVARCHAR2LIST('a','b', 'c');`
+```
+
+Both prompts will suggest a similar syntax to create an EPAS-compatible view:
+
+```
+CREATE OR REPLACE VIEW HRPLUS.VW_STRING_LIST (STRING_VAL) AS
+  SELECT UNNEST(ARRAY['a','b', 'c']) AS COLUMN_VALUE;
+```
+
+## Understanding an imprecise syntax error
+
+Sometimes, the Migration Portal provides an imprecise or vague error message for a syntax issue: 
+
+![Imprecise syntax error message](../../images/mp_vague_error.png)
+
+
+You can ask ask the AI Copilot to explain the issue: 
+
+```
+What is the issue with the following argument? `PIVOT ( COUNT(expense_type_id) FOR expense_type_id IN (10, 20, 30) ) ORDER BY employee_ref;`
+```
+
+And to suggest an equivalent: 
+
+```
+Can you provide a Postgres equivalent for this Oracle query? `PIVOT ( COUNT(expense_type_id) FOR expense_type_id IN (10, 20, 30) ) ORDER BY employee_ref;`
```

Comment:
I find it interesting that specifying just the portion of the SQL query statement starting from the PIVOT operator actually returns a useful solution.  Personally, I would think it better in general to specify the entire Select statement in this case and for other issues. For example, 
> Can you provide a Postgres compatible version of this Oracle query? 'SELECT EMPLOYEE_REF,"10","20","30" FROM ( SELECT employee_ref, expense_type_id FROM expenses ) PIVOT ( COUNT(expense_type_id) FOR expense_type_id IN (10, 20, 30) ) ORDER BY employee_ref;'